src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorConnectionContextTest.java

```diff
@@ -77,10 +78,30 @@ public void test_2_isNodeUnhealthyWithInvalidConnection_returnFalse() {
   public void test_3_isNodeUnhealthyExceedsFailureDetectionCount_returnTrue() {
     final int expectedFailureCount = FAILURE_DETECTION_COUNT + 1;
     context.setFailureCount(FAILURE_DETECTION_COUNT);
+    context.resetInvalidNodeStartTime();
 
-    context.setConnectionValid(false);
+    context.setConnectionValid(false, System.currentTimeMillis(), VALIDATION_INTERVAL_MILLIS);
 
-    Assertions.assertTrue(context.isNodeUnhealthy());
+    Assertions.assertFalse(context.isNodeUnhealthy());
     Assertions.assertEquals(expectedFailureCount, context.getFailureCount());
+    Assertions.assertTrue(context.isInvalidNodeStartTimeDefined());
+  }
+
+  @Test
+  public void test_4_isNodeUnhealthyExceedsFailureDetectionCount_returnTrue() {
```

Comment:
NIT: Is the "returnTrue" needed at the end, since we can see the assertTrue at the end?
Could be "test_nodeUnhealthyCountExceedsFailureCount" ?